### ISSUE TITLE
Refactor Nano **RtpPacket**

### ISSUE THREAD

**Issue by tuxuser** [OpenXbox/xbox-smartglass-csharp#38]
Currently every packet is represented as `RtpPacket` and individual segments of that packet are forwarded to inner layers:

Schema:
Transport Layer: RtpPacket
ChannelManager: ChannelControlData
VideoChannel: VideoData, ServerHandshake, ClientHandshake, ..
etc. etc.

Would be better to represent the `RtpPacket` as it's actual type already in **transport layer** and pass the entire object around, to deliver needed metadata (f.e. Marker bit to VideoChannel)

Long story short: refactoring needed.

**Comment by tuxuser** [comment]
Done
